src/components/UI/BookComponent/BookComponent.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  Card,
+  CardContent,
+  CardMedia,
+  Typography,
+  Button,
+} from "@mui/material";
+import classes from "./BookComponent.module.css";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Book } from "../../../Types/bookTypes";
+
+interface BookProps {
+  book: Book;
+}
+
+const BookComponent: React.FC<BookProps> = ({ book }) => {
+  const [isHovering, setIsHovering] = useState(false);
+  const navigate = useNavigate();
+
+  const handleMouseEnter = () => setIsHovering(true);
+  const handleMouseLeave = () => setIsHovering(false);
+
+  const handleClick = () => navigate(`/books/${book.id}`);
+
+  return (
+    <Card
+      className={classes.book}
+      style={{ boxShadow: "none" }}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+      onClick={handleClick}
+    >
+      <CardMedia
+        className={classes.cover}
+        component="img"
+        image={book.cover}
+        alt={book.name}
+      />
+      {isHovering && (
+        <Button
+          className={classes.addButton}
+          style={{
+            position: "absolute",
+            transition: "opacity 0.3s ease-in-out",
```

Comment:
I see you made the transition duration here, please remove the redundancy in the CSS or remove it from here and depend on CSS (recommended)

Reply:
it was there but mui styles overrode it